fasterer.gemspec

```diff
@@ -20,6 +20,7 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency 'colorize', '~> 0.7'
   spec.add_dependency 'ruby_parser', '~> 3.11.0'
+  spec.add_dependency 'homecoming', '~> 0.1.1'
```

Comment:
Rubocop also uses this gem or does it tackle it in some other way? I don't want to add more dependencies if not needed of course.

Reply:
Rubocop has a [file finder](https://github.com/rubocop-hq/rubocop/blob/cb62bafc4e109987c8c5d546edf6250078038aad/lib/rubocop/file_finder.rb) that adds more features like looking in your home dir, which is helpful. 

Looking around some more, I see [reek](https://github.com/troessner/reek) does [the same thing](https://github.com/troessner/reek/blob/ca2cebcee7f69ee7fd6c1bb6143ee42f2d4f380e/lib/reek/configuration/configuration_file_finder.rb). They also have more [desired functionality](https://github.com/troessner/reek/issues/1342) on their radar which would be of benefit.

It seems there is some duplication in the "code quality config file resolution" space. I wonder if the owners of all those projects would appreciate PRs to use a gem that does this, either via an updated [homecoming](https://github.com/rrrene/homecoming) gem or one leveraging it. Absolute or relative path specification (like rails-best-practices allows), then upwards traversal, then look in home dir, returning an array of all matches. I do not propose it does the config file merging that rubocop does. 

@rrrene, @mvz, @bbatsov @DamirSvrtan, is this worth exploring?

Reply:
@hlascelles You can have the `homecoming` feature by using these 6 lines of code:

```ruby
require 'pathname'

filename = ".fasterer.yml"
Pathname(Dir.pwd).ascend.map do |dir|
  f = File.join(dir, filename)
  File.exist?(f) ? f : nil
end.compact.reverse
```

`homecoming` was created before `Pathname` was added to Ruby and was kept to ensure compatibility for Ruby versions `< 1.9.3`. There's probably no real need for it today. :+1:

Reply:
Excellent, thank you. Just a question now as to whether a gem is desired to bring together the additional locations in a consistent way (specified switch, home dir, etc...) across quality tooling gems?

Reply:
Hey Harry!

Thank you for taking a great initiative and research on this! Also thanks @rrrene for the code snippet!

I'd wait for other code quality tool maintainers to voice their opinions on an initiative such as you proposed (which I'm totally up for)! I would merge a small utility gem like that into fasterer. 

I've never heard of homecoming before and @rrrene's snippet seems pretty cool, but I'd want it tested. If other maintainers are open for merging in a small utility gem like that, I'd wait for it to be made. 

If not, I'd replace homecoming with the code proposed by @rrrene and leave the tests as you've wrote them. Ideally the tests should keep passing no matter what the implementation is. How does that sound to you @hlascelles?

Btw it was really nice meeting you at Paris.rb!

Reply:
The code that does the directory traversal in Reek is already very simple, so I'm not sure a gem would help much. Especially since there will be subtle differences in how things are done that such a gem would have to account for.